src/tests/rtl/suite/time_conversion.c

```diff
@@ -0,0 +1,169 @@
+        { .time_fields = { .Year = 1601, .Month = 1, .Day = 1, .Second = 59 }, .result_expected = TRUE, .time_expected.QuadPart = 590000000 },
+        { .time_fields = { .Year = 1601, .Month = 1, .Day = 1, .Millisecond = 1000 }, .result_expected = FALSE },
+        { .time_fields = { .Year = 1601, .Month = 1, .Day = 1, .Millisecond = 999 }, .result_expected = TRUE, .time_expected.QuadPart = 9990000 },
+        // Check that leap year is working as intended for both fail and success
+        { .time_fields = { .Year = 1603, .Month = 2, .Day = 29 }, .result_expected = FALSE },
+        { .time_fields = { .Year = 1603, .Month = 2, .Day = 28 }, .result_expected = TRUE, .time_expected.QuadPart = 680832000000000 },
+        { .time_fields = { .Year = 1604, .Month = 2, .Day = 30 }, .result_expected = FALSE },
+        { .time_fields = { .Year = 1604, .Month = 2, .Day = 29 }, .result_expected = TRUE, .time_expected.QuadPart = 997056000000000 },
+        // Check large integer max (max is actually 9223372036854770000 since last 4 digits are always 0.)
+        { .time_fields = { .Year = 30828, .Month = 9, .Day = 14, .Hour = 2, .Minute = 48, .Second = 5, .Millisecond = 477 }, .result_expected = TRUE, .time_expected.QuadPart = 9223372036854770000 },
+        // Basically overflow max test
+        { .time_fields = { .Year = 30828, .Month = 9, .Day = 14, .Hour = 2, .Minute = 48, .Second = 5, .Millisecond = 478 }, .result_expected = TRUE, .time_expected.QuadPart = -9223372036854771616 },
+    };
+    enum { time_tests_size = ARRAY_SIZE(time_tests) };
+
+    for (unsigned i = 0; i < time_tests_size; i++) {
+        // NOTE: PTIME_FIELDS cast is used to bypass warning from compiler
+        time_tests[i].result_actual = RtlTimeFieldsToTime((PTIME_FIELDS)&time_tests[i].time_fields, &time_tests[i].time_actual);
+    }
+    GEN_CHECK_ARRAY_MEMBER(time_tests, time_actual.QuadPart, time_expected.QuadPart, time_tests_size, "time_tests");
+    GEN_CHECK_ARRAY_MEMBER(time_tests, result_actual, result_expected, time_tests_size, "time_tests");
+
+    print_test_footer(func_num, func_name, test_passed);
+}
+
+void test_RtlTimeToTimeFields()
+{
+    const char* func_num = "0x0131";
+    const char* func_name = "RtlTimeToTimeFields";
+    BOOL test_passed = 1;
+    print_test_header(func_num, func_name);
+
+    TIME_FIELDS time_fields;
+    LARGE_INTEGER time;
+
+    // Test #1 - expected range of acceptable and unacceptable values
+    typedef struct _time_fields_test {
+        const TIME_FIELDS expected;
+        const LARGE_INTEGER time;
+        TIME_FIELDS actual;
+    } time_fields_test;
+
+    time_fields_test time_fields_tests[] = {
+        { .expected = { .Year = 1601, .Month = 1, .Day = 1 }, .time.QuadPart = 0 },
+        { .expected = { .Year = 1601, .Month = 1, .Day = 1, .Millisecond = 1 }, .time.QuadPart = 10000 },
+        { .expected = { .Year = 1601, .Month = 1, .Day = 1, .Second = 1 }, .time.QuadPart = 10000000 },
```

Comment:
nit: Why not keep comments from test 1 - 3 you had? Then 4th test's comment can say extended test or whatever.

Reply:
closing nitpicker since said nah.